src/lib/operators/join_hash.cpp

```diff
@@ -685,6 +688,27 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
                              _right_in_table->get_chunk(ChunkID{0})->get_column(ColumnID{0}))
                              ? true
                              : false;
+    auto only_output_right_input = !_inputs_swapped || (_mode != JoinMode::Semi && _mode != JoinMode::Anti);
+
+    /**
+     * Two Caches to avoid redundant reference materialization for Reference input tables. As the there might be
+     *  quite a lot Partitions (>500 seen), input Chunks (>500 seen), and columns (>50 seen), this speeds up
+     *  write_output_chunks a lot()
+     *
+     * They do two things:
+     *      - Make it possible to re-use output pos lists if two columns in the input table have exactly the same
+     *          PosLists Chunk by Chunk
+     *      - Avoid creating the std::vector<const PosList*> for each Partition over and over again.
+     */
+    PosListsByColumn left_pos_lists_by_column;
```

Comment:
`    // This holds one entry per column in the table, not per BaseColumn in a single chunk`

Reply:
yap